### PR TITLE
[NEW] Show server history

### LINKED ISSUE #2348: [IMPROVEMENT] Save Server URL
- Your Rocket.Chat Experimental app version: 4.9.0.2132
- Your Rocket.Chat server version: 3.4.2
- Device (or Simulator) you're running with: Any

Hi,

I'm not sure if this was requested specifically for the mobile app already, but could you maybe implement a possibility to save the server URL on the login screen?
Our users are already unhappy about the forced logout every X days. Additionaly they have to type in the URL every single time they get logged out. 
I'm not asking for anything fancy. Just a checkbox "save URL" or something on the login screen. Would be great :)

Thank you in advance

### PR DESCRIPTION
**Pull request by EzequielDeOliveira**
## Proposed changes

Shows a list of previous servers/users, so the user can login again on previous users quickly.

## Issue(s)	

Closes #2348 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by EzequielDeOliveira** [comment]
My bad, I'll take a look at this problem!

**Review by djorkaeffalexandre — CHANGES_REQUESTED** [review]

File: app/views/NewServerView/ServerInput/Item.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import { themes } from '../../../constants/colors';
+import { CustomIcon } from '../../../lib/Icons';
+import sharedStyles from '../../Styles';
+import Touch from '../../../utils/touch';
+
+const styles = StyleSheet.create({
+	container: {
+		height: 56,
+		paddingHorizontal: 15,
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/views/NewServerView/ServerInput/Item.js:16]: We need to refactor this kind of content soon.
Currently, most of the horizontal content paddings are 15.
Example: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/Styles.js#L11

Right now, it's better to keep this bad implementation.

**Review by diegolmello — APPROVED** [review]
LGTM

**Comment by didierm** [comment]
@EzequielDeOliveira Thank you so much !
This allows us to force session expiration on mobile.

**Comment by aenonGit** [comment]
@EzequielDeOliveira @diegolmello am I the only one experiecing crashes? On the open server, where I have a username/password account everything goes well but on my own server with custom oauth, when I click the server suggestion the app crashes

**Comment by diegolmello** [comment]
> @EzequielDeOliveira @diegolmello am I the only one experiecing crashes? On the open server, where I have a username/password account everything goes well but on my own server with custom oauth, when I click the server suggestion the app crashes

@aenonGit Search for opened issues and open a new one (if you didn't find it) with detailed info is the only way to help you. PRs appreciated.

**Comment by devZer0** [comment]
the auto-logout with missing workspace url/history is a real bummer and a major reason why our community has problems accepting rocketchat as web/team chat.  

when will this fix be available in the app/play-store ?

**Comment by diegolmello** [comment]
@devZer0 it's available on 4.11

**Comment by devZer0** [comment]
mhh, ok, i missed that. you are right. sorry. 

but i think this could be solved more intelligently by showing the last used server in the dialogue _before_ the dialogue for entering that url another time or selecting the old one from the history.  

most non-technical users will be overchalleged by this when the suddenly being thrown out of the chat and need to re-enter the server url.  many won't try to re-type something they don't know and most of our users in our community will not know that url weeks after usage and need to ask for it another time.

it's like throwing someone out of a house with a thousand doors with some remark way down on the pavement, which door he needs to choose for re-entering the house.

**Comment by diegolmello** [comment]
Got it, but it is redirecting you automatically after server logs you out.
Maybe you can update, test and give your feedback again later.
WDYT?

**Comment by devZer0** [comment]
sounds promising - but i have 4.11 and it didn't.  i cannot force server-side login timeout (as i don't own this server) for retry, so need to wait for another occurence...

**Comment by diegolmello** [comment]
Ok. Open a new issue when it happens with details.
A video would be good, just to be sure we're talking about the same thing.

**Pull request merged**